### DISCUSSION TITLE
compiling error on cento 7 with devtoolset-7-gcc-gfortran

### DISCUSSION THREAD

**Discussion by john-zhao-git** [OpenRadioss/OpenRadioss#511] [Q&A] [answered]
hey guys, 

I got following error when compiling the code on centos 7 with devtoolset-7-gcc-gfortran

[ 99%] Building CXX object CMakeFiles/starter_linux64_gf.dir/apps/OpenRadioss/common_source/tools/graphs/Graph_api.cpp.o
[ 99%] Building CXX object CMakeFiles/starter_linux64_gf.dir/apps/OpenRadioss/common_source/tools/graphs/Graph.cpp.o
[100%] Building CXX object CMakeFiles/starter_linux64_gf.dir/apps/OpenRadioss/common_source/tools/container/IntVector_api.cpp.o
Linking Fortran executable starter_linux64_gf
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld:/apps/OpenRadioss/starter/../extlib/metis/linux64/libmetis_linux64_gcc.a: file format not recognized; treating as linker script
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld:/apps/OpenRadioss/starter/../extlib/metis/linux64/libmetis_linux64_gcc.a:1: syntax error
collect2: error: ld returned 1 exit status
make[2]: *** [starter_linux64_gf] Error 1
make[1]: *** [CMakeFiles/starter_linux64_gf.dir/all] Error 2
make: *** [all] Error 2
 
what did I miss? 

Thanks
John

**Comment by OlivierWienholtz** [comment, marked as answer]
Hi John

I can see 2 causes for your issue : 
1/
have you installed git-lfs ? 
the libraries are handled using git-lfs

2/ The minimum Devtoolset is 8 to build Radioss. 
You'll meet link issues due to runtime incompatibilities with the librairies we ship.

Hope it helps 
Olivier

**Reply by john-zhao-git** [reply]
Olivier,

Thank you! I installed devtoolset 8, and I got same error....
[100%] Building CXX object CMakeFiles/starter_linux64_gf.dir/apps/OpenRadioss/common_source/tools/container/IntVector_api.cpp.o
Linking Fortran executable starter_linux64_gf
/opt/rh/devtoolset-8/root/usr/libexec/gcc/x86_64-redhat-linux/8/ld:/apps/OpenRadioss/starter/../extlib/metis/linux64/libmetis_linux64_gcc.a: file format not recognized; treating as linker script
/opt/rh/devtoolset-8/root/usr/libexec/gcc/x86_64-redhat-linux/8/ld:/apps/OpenRadioss/starter/../extlib/metis/linux64/libmetis_linux64_gcc.a:1: syntax error
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/starter_linux64_gf.dir/build.make:34072: starter_linux64_gf] Error 1
make[1]: *** [CMakeFiles/Makefile2💯 CMakeFiles/starter_linux64_gf.dir/all] Error 2
make: *** [Makefile:117: all] Error 2

can you elaborate on git-lfs? what do I need to do with git-lfs to resolve this issue?

Thanks

**Reply by john-zhao-git** [reply]
I installed git-lfs and devtoolset-8, still getting same error

[100%] Building CXX object CMakeFiles/starter_linux64_gf.dir/apps/OpenRadioss/common_source/tools/container/IntVector_api.cpp.o
Linking Fortran executable starter_linux64_gf
/opt/rh/devtoolset-8/root/usr/libexec/gcc/x86_64-redhat-linux/8/ld:/apps/OpenRadioss/starter/../extlib/metis/linux64/libmetis_linux64_gcc.a: file format not recognized;
/opt/rh/devtoolset-8/root/usr/libexec/gcc/x86_64-redhat-linux/8/ld:/apps/OpenRadioss/starter/../extlib/metis/linux64/libmetis_linux64_gcc.a:1: syntax error
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/starter_linux64_gf.dir/build.make:34072: starter_linux64_gf] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/starter_linux64_gf.dir/all] Error 2
make: *** [Makefile:117: all] Error 2
 
 
[root@ip-10-0-74-18 starter]# which git-lfs
/bin/git-lfs


how can I resolve this?

**Reply by john-zhao-git** [reply]
Issue resolved after download clean copy of the code, compiled with devtoolset 11 

Thanks!
